src/journal/models.py

```diff
@@ -592,46 +600,62 @@ def is_serial(self):
         else:
             return False
 
-    def update_display_title(self):
-        if self.issue_type and self.issue_type.code != 'issue':
-            self.display_title = self.issue_title
+    @cached_property
+    def display_title(self):
+        return mark_safe(
+            self.cached_display_title
+            or self.update_display_title(save=True)
+        )
+
+    def update_display_title(self, save=False):
+        title = None
+        if self.issue_type and self.issue_type.code == 'issue':
+            if save:
+                self.save()
+                return self.cached_display_title
+            title = self.cached_display_title = self.pretty_issue_identifier
         else:
-            self.display_title = self.pretty_issue_identifier
+            self.cached_display_title = ''
+            title = self.issue_title
+
+        return title
 
     def issue_title_parts(self, article=None):
         journal = self.journal
         volume = issue = year = issue_title = article_number = page_numbers = ''
 
         if journal.display_issue_volume and self.volume:
-            volume = ugettext("Volume") + " {}".format(self.volume)
+            volume = "{%% trans 'Volume' %%} %s" % self.volume
         if journal.display_issue_number and self.issue and self.issue != "0":
             issue = ugettext("Issue") + " {}".format(self.issue)
+            issue = "{%% trans 'Issue' %%} %s" % self.issue
         if journal.display_issue_year and self.date:
             year = "{}".format(self.date.year)
         if journal.display_issue_title:
             issue_title = self.issue_title
         if journal.display_article_number and article and article.article_number:
-            article_number = ugettext("Article") + " {}".format(article.article_number)
+            article_number = "{%% trans 'Article' %%} %s" % article.article_number
         if journal.display_article_page_numbers and article:
             if article.page_range:
                 page_numbers = article.page_range
             elif article.total_pages:
-                if article.total_pages != 1:
-                    label = ugettext('pages')
-                else:
-                    label = ugettext('page')
                 num_pages = str(article.total_pages)
-                page_numbers = f'{num_pages} {label}'
+                page_numbers = f"{num_pages} {{% trans 'label' %}}"
 
         return [volume, issue, year, issue_title, article_number, page_numbers]
```

Comment:
With the changes in this function there is a failing test in submission: `submission.tests.SubmissionTests.test_article_issue_title`. 

And the function it tests, `article.issue_title`, needs some changes:
![Screenshot from 2022-08-04 11-13-31](https://user-images.githubusercontent.com/47217308/182823309-385b98f2-8cf7-41d4-98af-2c2981298eae.png)